tests/drivers/sensor/accel/src/main.c

```diff
@@ -11,72 +11,312 @@
+	zassert_ok(sensor_get_scale(dev, SENSOR_TYPE_GYROSCOPE, &scale_metadata[1]), NULL);
+	zassert_equal(SENSOR_RANGE_UNITS_ACCEL_G, scale_metadata[0].range_units, NULL);
+	zassert_equal(SENSOR_RANGE_UNITS_ANGLE_DEGREES, scale_metadata[1].range_units, NULL);
+
+	/* Test converted values in SI units */
+	sensor_sample_to_three_axis_data(&scale_metadata[0], data, 0);
+	zassert_within(FLOAT_TO_FP(0.0f), data->readings[0].x, FLOAT_TO_FP(0.01f),
+		       "Expected %" PRIf " to be within %" PRIf " of %" PRIf,
+		       PRIf_ARG(FLOAT_TO_FP(0.0f)), PRIf_ARG(FLOAT_TO_FP(0.01f)),
+		       PRIf_ARG(data->readings[0].x));
+	zassert_within(FLOAT_TO_FP(0.1f), data->readings[0].y, FLOAT_TO_FP(0.01f),
+		       "Expected %" PRIf " to be within %" PRIf " of %" PRIf,
+		       PRIf_ARG(FLOAT_TO_FP(0.1f)), PRIf_ARG(FLOAT_TO_FP(0.01f)),
+		       PRIf_ARG(data->readings[0].y));
+	zassert_within(FLOAT_TO_FP(0.2f), data->readings[0].z, FLOAT_TO_FP(0.01f),
+		       "Expected %" PRIf " to be within %" PRIf " of %" PRIf,
+		       PRIf_ARG(FLOAT_TO_FP(0.2f)), PRIf_ARG(FLOAT_TO_FP(0.01f)),
+		       PRIf_ARG(data->readings[0].z));
+
+	zassert_ok(sensor_read_data(dev, SENSOR_TYPE_GYROSCOPE, data), NULL);
+	zassert_equal(0x0b01, data->readings[0].x, NULL);
+	zassert_equal(0x0eac, data->readings[0].y, NULL);
+	zassert_equal(0x1257, data->readings[0].z, NULL);
 
-static void test_sensor_accel_basic(const struct device *dev)
+	/* Test converted values in SI units */
+	sensor_sample_to_three_axis_data(&scale_metadata[1], data, 0);
+	zassert_within(FLOAT_TO_FP(171.9f), data->readings[0].x, FLOAT_TO_FP(0.1f),
+		       "Expected %" PRIf " to be within %" PRIf " of %" PRIf,
+		       PRIf_ARG(FLOAT_TO_FP(171.9f)), PRIf_ARG(FLOAT_TO_FP(0.1f)),
+		       PRIf_ARG(data->readings[0].x));
+	zassert_within(FLOAT_TO_FP(229.2f), data->readings[0].y, FLOAT_TO_FP(0.1f),
+		       "Expected %" PRIf " to be within %" PRIf " of %" PRIf,
+		       PRIf_ARG(FLOAT_TO_FP(229.2f)), PRIf_ARG(FLOAT_TO_FP(0.1f)),
+		       PRIf_ARG(data->readings[0].y));
+	zassert_within(FLOAT_TO_FP(286.6f), data->readings[0].z, FLOAT_TO_FP(0.1f),
+		       "Expected %" PRIf " to be within %" PRIf " of %" PRIf,
+		       PRIf_ARG(FLOAT_TO_FP(286.6f)), PRIf_ARG(FLOAT_TO_FP(0.1f)),
+		       PRIf_ARG(data->readings[0].z));
+}
+
+ZTEST_F(sensor_accel, test_accel_0_read_data) { test_sensor_accel_basic(fixture->accel_0); }
+
+ZTEST_F(sensor_accel, test_accel_1_read_data) { test_sensor_accel_basic(fixture->accel_1); }
+
+ZTEST_F(sensor_accel, test_set_bias)
 {
-	zassert_equal(sensor_sample_fetch(dev), 0, "fail to fetch sample");
+	int8_t bias[3];
 
-	for (int i = 0; i < ARRAY_SIZE(channel); i++) {
-		struct sensor_value val;
+	zassert_ok(sensor_set_bias(fixture->accel_1, SENSOR_TYPE_ACCELEROMETER, INT16_C(20),
+				   FLOAT_TO_FP(4.0f), FLOAT_TO_FP(-8.0f), FLOAT_TO_FP(12.0f), true),
+		   NULL);
+	zassert_ok(bmi160_emul_get_bias(fixture->accel_emul_1, SENSOR_TYPE_ACCELEROMETER, &bias[0],
```

Comment:
I'm not totally grokking the difference between this test and `test_get_bias`. But wouldn't a more direct test of setting the bias be that the sensor readings are appropriately offset from their nominal values?